websites/S/Stremio/presence.ts

```diff
@@ -0,0 +1,201 @@
+					"[class='ng-scope ng-binding selected']"
+				)?.textContent;
+				if (!genreSort) {
+					genreSort =
+						document.querySelectorAll(
+							"[class='ng-binding ng-scope selected']"
+						)[2]?.textContent ?? "None";
+				}
+				if (genreSort === "Top") genreSort = "All";
+				if (buttons) {
+					presenceData.buttons = [
+						{
+							label: `Browse: ${active}`,
+							url: document.location.href
+						}
+					];
+				}
+				presenceData.details = `Browsing ${active}`;
+				presenceData.state = `Genre: ${genreSort}`;
+			} else presenceData.state = "Browsing Movies";
+		} else if (page.includes("/library")) {
+			genreSort = document
+				.querySelector<HTMLSelectElement>(
+					"#library > div.sort-filter > div.custom-select.lib-sort > select"
+				)
+				?.value.replace("SORT", "")
+				.replace("_", " ")
+				.trim();
+			if (genreSort) {
+				presenceData.state = `Sorted by: ${genreSort.substring(0, 1)}${genreSort
+					.substring(1, genreSort.length)
+					.toLowerCase()}`;
+			}
+			if (active) {
+				presenceData.details = `${active} Library`;
+				if (buttons) {
+					presenceData.buttons = [
+						{
+							label: `View ${active} Library`,
+							url: document.location.href
+						}
+					];
+				}
+			} else {
+				if (buttons) {
+					presenceData.buttons = [
+						{
+							label: "View Library",
+							url: document.location.href
+						}
+					];
+				}
+				presenceData.details = "Library";
+			}
+		} else if (page.includes("/calendar")) {
+			if (buttons) {
+				presenceData.buttons = [
+					{
+						label: "View Calendar",
+						url: document.location.href
+					}
+				];
+			}
+			presenceData.details = "Calendar";
+		} else if (page.includes("player")) {
+			if (video?.duration) {
+				timestamp = presence.getTimestampsfromMedia(video);
+				pauseCheck = video.paused;
+			} else if (!video.duration && document.querySelector("#controlbar-top")) {
+				const split = document
+					.querySelector("#play-progress-text")
+					?.textContent.split("/");
+				if (split?.[0]) {
+					timestamp = presence.getTimestamps(
+						presence.timestampFromFormat(split[0].trim()),
+						presence.timestampFromFormat(split[1].trim())
+					);
+				} else timestamp = void 0;
+
+				pauseCheck = document
+					.querySelector("#controlbar-top")
+					?.firstElementChild?.className.includes("pause")
+					? false
+					: true;
+			}
+			delete presenceData.startTimestamp;
+			if (
+				!pauseCheck &&
+				!document.querySelector("#loading-logo").className.includes("flashing")
+			) {
+				presenceData.endTimestamp = timestamp[1] ?? 0;
+				presenceData.smallImageKey = "play";
+			} else {
+				delete presenceData.endTimestamp;
+				presenceData.smallImageKey = "pause";
+			}
+			title = document
+				.querySelector("head > title")
+				?.textContent.replace("Stremio -", "");
+			presenceData.details = title;
+			if (buttons) {
+				presenceData.buttons = [
+					{
+						label: `Join View Party For: ${title}`,
```

Comment:
no colon after for, also don't repeat the title
```suggestion
						label: "Join View Party",
```